app/models/ability.rb

```diff
@@ -5,6 +5,8 @@ def initialize(user)
     if user
       can %i(create read destroy update), Wishlist, user_id: user.id
       can :update, Item, wishlist: { user_id: user.id }
+    else
+      can %i(reserve), Item
```

Comment:
No need for the `else`. It can be put outside the whole `if`. What's more you can delete completely, and not try to authorize this if anyone can do this.